doc/development/guide/documentation.rst

```diff
@@ -1,39 +1,449 @@
+are used to document the underlying implementation. Use comments where applicable to make
+your code easy to follow and understand, keeping to the following guidelines.
+
+* **Use comments to explain the implementation or algorithm, never to describe the code.**
+  Assume the reader has basic understanding of common programming principles and syntax.
+  Do not assume the reader knows what you're trying to do with it!
+
+  .. note::
+
+      If additional implementation details have been requested by reviewers during code review,
+      these must be incorporated as code comments.
+
+* **Code should be self-documented where possible.** Code should be clear and concise,
+  with the logical flow easily followed by the reader. This can be done by using common
+  programming patterns alongside sensible variable and function names.
+
+  Note that this does not exclude the use of code comments; comments are particularly valuable
+  in quantum software, where the underlying algorithm can be high non-trivial. Rather,
+  self-documented code simply excludes *unnecessary* comments.
+
+* **Comments should be as close to the described code as possible.** These can either be
+  single- or multi-line comments *above* the described code block,
+
+  .. code-block:: python3
+
+      # Note: in the following template definition, we pass the observable, measurement,
+      # and wires as *default arguments* to named parameters. This is to avoid
+      # Python's late binding closure behaviour
+      # (see https://docs.python-guide.org/writing/gotchas/#late-binding-closures)
+
+      def circuit(params, _obs=obs, _m=m, _wires=wires, **kwargs):
+          template(params, wires=_wires, **kwargs)
+          return MEASURE_MAP[_m](_obs)
+
+  or, for shorter comments, single line comments at the end of the line.
+
+* **Avoid markup and complex formatting in comments.** Markup, such as text formatting and
+  hyperlink markup, are unnecessary in comments as they are not rendered, and will simply
+  be read in plain text. Use text formatting sparingly for emphasis, and simply insert URLs
+  directly. In addition, avoid complex formatting such as tables --- these are difficult
+  to maintain and modify.
+
+
+Punctuation and spelling
+------------------------
+
+* Use Canadian spellings. That is, "centre" instead of "center", "realize" instead of "realise",
+  "acknowledgement" instead of "acknowledgment", "colour" instead of "color", etc.
+
+* Use complete sentences with capitalization and punctuation. The one exemption
+  is when describing arguments, return values, attributes, and exceptions; sentence fragments
+  may be used for short descriptions:
+
+  .. code-block:: rest
+
+      wires (List[int]): subsystems the operation is applied to
+      diff_method (str or None): the method of differentiation to use in the created QNode
+
+  Longer argument descriptions should use capitalization and punctuation:
```

Comment:
It's not necessary "length" that's important, but when the description is more naturally given as multiple sentences, then we prefer that (rather than having awkward semicolon schemes).
e.g.,
```some_arg (int): The argument for the thing. The convention used is ...```

Reply:
>  Multi-sentence argument descriptions, or longer sentence fragments with mid-sentence punctuation marks, should use full capitalization and punctuation: